### PR TITLE
Changes to allow specific files to be checked

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Changes to allow only a terraform apply to run if a specific file has been changed, override option included - if set to $true it will ignore the files check and run terraform apply

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by thomast1906** [comment]
set label to do not merge. Noticing an Azure DevOps issue - although on my test environment it is not replicating that issue..

File: pipeline-templates/terraform.yaml [hunk 1]
```diff
@@ -29,9 +29,18 @@
...
+        displayName: "CheckFileChanges"
+        env:
+          files_list: '${{ parameters.files_list }}'
+        inputs:
+          filePath: pipeline-scripts/check-changes.sh
+
       - task: TerraformCLI@0
         displayName: Apply - ${{ parameters.environment }} - ${{ parameters.component }}
-        condition: and(succeeded(), eq(variables.isMaster, true))
+        condition: or(and(succeeded(), eq(variables['CheckFileChanges.SOURCE_CODE_CHANGED'], 'true')),and(succeeded(), eq(${{ parameters.override }}, 'true')))
```
Comment (adusumillipraveen) [inline pipeline-templates/terraform.yaml:43]: can it be ?

condition: and(succeeded(), or(eq(variables['CheckFileChanges.SOURCE_CODE_CHANGED'], 'true'),eq(${{ parameters.override }}, 'true')))

File: azure_pipeline-global.yaml [hunk 2]
```diff
@@ -39,6 +39,8 @@ parameters:
       storage_account_rg: 'core-infra-sbox-rg'
       storage_account_name: 'cftappssbox'
       dependsOn: 'Precheck'
+      files_list: 'components/global modules/azure-landing-zone environments/sbox/sbox.tfvars environments/variables.tf'
+      override: false
```
Comment (adusumillipraveen) [inline azure_pipeline-global.yaml:43]: Can we rename this var to be more obvious ? something like alwaysApply

Comment (adusumillipraveen) [reply azure_pipeline-global.yaml:43]: or we could change it to say, if there is no file_list- apply it always.  so that we dont need a new var

**Review by adusumillipraveen — APPROVED** [review]
LGTM, added a couple of suggestions

File: pipeline-templates/terraform.yaml (hunk 1 above)
Comment (thomast1906) [reply pipeline-templates/terraform.yaml]: Good shout - now added

File: azure_pipeline-global.yaml (hunk 2 above)
Comment (thomast1906) [reply azure_pipeline-global.yaml:43]: I can change to alwaysApply , regarding the check for no file_list - possible but if a user wants to apply when a file in that list hasn't changed.

will keep the true/false flag and ammend to alwaysApply

**Pull request merged**